version.props

```diff
@@ -1,22 +1,25 @@
 <Project>
   <PropertyGroup>
-    <VersionPrefix>2.1.0</VersionPrefix>
-    <VersionSuffix>rtm</VersionSuffix>
+    <AspNetCoreMajorVersion>2</AspNetCoreMajorVersion>
+    <AspNetCoreMinorVersion>1</AspNetCoreMinorVersion>
+    <AspNetCorePatchVersion>1</AspNetCorePatchVersion>
+    <VersionPrefix>$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).$(AspNetCorePatchVersion)</VersionPrefix>
+    <PrereleaseVersionLabel>rtm</PrereleaseVersionLabel>
 
     <ExperimentalVersionPrefix>0.1.0</ExperimentalVersionPrefix>
     <ExperimentalVersionSuffix>rtm</ExperimentalVersionSuffix>
 
+    <PackageVersionSuffix Condition=" '$(IsFinalBuild)' != 'true' AND '$(PrereleaseVersionLabel)' != '' ">$(PrereleaseVersionLabel)</PackageVersionSuffix>
+    <PackageVersionSuffix Condition=" '$(IsFinalBuild)' != 'true' AND '$(BuildNumber)' != '' ">$(PackageVersionSuffix)-$(BuildNumber)</PackageVersionSuffix>
+    <PackageVersionSuffix Condition=" '$(IsFinalBuild)' == 'true' AND '$(PrereleaseVersionLabel)' != 'rtm' ">$(PrereleaseVersionLabel)-final</PackageVersionSuffix>
+    <!-- NB: VersionSuffix is empty if '$(IsFinalBuild)' == 'true' AND '$(PrereleaseVersionLabel)' == 'rtm' -->
 
-    <PackageVersion Condition=" '$(IsFinalBuild)' == 'true' AND '$(VersionSuffix)' == 'rtm' ">$(VersionPrefix)</PackageVersion>
-    <PackageVersion Condition=" '$(IsFinalBuild)' == 'true' AND '$(VersionSuffix)' != 'rtm' ">$(VersionPrefix)-$(VersionSuffix)-final</PackageVersion>
-
-    <VersionSuffix Condition=" '$(VersionSuffix)' != '' AND '$(BuildNumber)' != '' ">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
-
-    <PackageVersion Condition=" '$(IsFinalBuild)' != 'true' ">$(VersionPrefix)</PackageVersion>
-    <PackageVersion Condition=" '$(IsFinalBuild)' != 'true' AND '$(VersionSuffix)' != '' ">$(VersionPrefix)-$(VersionSuffix)</PackageVersion>
+    <PackageVersion>$(VersionPrefix)</PackageVersion>
```

Comment:
Slight behavioural difference here so I'd like to double check.

Before:

| --- | IsFinalBuild = false | IsFinalBuild = true |
| --- | --- | --- |
| suffix = preview | PackageVersion: 2.1.0-preview1-30245, VersionSuffix: preview1-30245 | PackageVersion: 2.1.0-preview1-final, VersionSuffix: preview1-30245 |
| suffix = rtm | PackageVersion: 2.1.0-rtm-30245, VersionSuffix: rtm-30245 | PackageVersion: 2.1.0, VersionSuffix: rtm-30245 |

After:

| --- | IsFinalBuild = false | IsFinalBuild = true |
| --- | --- | --- |
| PrereleaseVersionLabel = preview | PackageVersion: 2.1.0-preview1-30245, VersionSuffix: preview1-30245 | PackageVersion: 2.1.0-preview1-final, VersionSuffix: preview1-final |
| PrereleaseVersionLabel = rtm | PackageVersion: 2.1.0-rtm-30245, VersionSuffix: rtm-30245 | PackageVersion: 2.1.0, VersionSuffix: |